### PR TITLE
Expose more QB actions to no-data users

### LINKED ISSUE #16836: Data export of native query not working if columns are reorganized without query permissions
**Describe the bug**
If the user doesn't have query permissions and reorganizes the columns of a saved native SQL query, the exports will contain a "missing-required-permissions" error. User is able to export fine if they do not change column order.

**Logs**
```js
{"database_id":2,"started_at":"2021-06-30T16:04:57.42403Z","error_type":"missing-required-permissions","json_query":{"type":"native","middleware":{"skip-results-metadata?":true,"format-rows?":false},"native":{"template-tags":{"AgingDate":{"id":"8205f989-c3c6-082b-b729-74d01d753d17","name":"AgingDate","display-name":"Aging Date","type":"date","required":false},"StartVendor":{"id":"2f0ff4db-35b4-ff3f-cc99-c60f71102a4f","name":"StartVendor","display-name":"Vendor Start #","type":"text"},"EndVendor":{"id":"8824e03b-a908-78c8-7b3b-c3ca85e45be2","name":"EndVendor","display-name":"Vendor End #","type":"text"}},"query":"<snip for privacy>"},"database":2,"async?":true,"cache-ttl":13},"status":"failed","class":"class clojure.lang.ExceptionInfo","stacktrace":["--> query_processor.middleware.permissions$perms_exception.invokeStatic(permissions.clj:27)","query_processor.middleware.permissions$perms_exception.doInvoke(permissions.clj:22)","query_processor.middleware.permissions$perms_exception.invokeStatic(permissions.clj:24)","query_processor.middleware.permissions$perms_exception.invoke(permissions.clj:22)","query_processor.middleware.permissions$fn__46594$check_ad_hoc_query_perms__46599$fn__46603.invoke(permissions.clj:60)","query_processor.middleware.permissions$fn__46594$check_ad_hoc_query_perms__46599.invoke(permissions.clj:49)","query_processor.middleware.permissions$fn__46631$check_query_permissions_STAR___46636$fn__46637.invoke(permissions.clj:71)","query_processor.middleware.permissions$fn__46631$check_query_permissions_STAR___46636.invoke(permissions.clj:64)","query_processor.middleware.permissions$check_query_permissions$fn__46650.invoke(permissions.clj:80)","query_processor.middleware.pre_alias_aggregations$pre_alias_aggregations$fn__48768.invoke(pre_alias_aggregations.clj:40)","query_processor.middleware.cumulative_aggregations$handle_cumulative_aggregations$fn__46851.invoke(cumulative_aggregations.clj:60)","query_processor.middleware.resolve_joined_fields$resolve_joined_fields$fn__49067.invoke(resolve_joined_fields.clj:102)","query_processor.middleware.resolve_joins$resolve_joins$fn__49380.invoke(resolve_joins.clj:171)","query_processor.middleware.add_implicit_joins$add_implicit_joins$fn__44925.invoke(add_implicit_joins.clj:190)","query_processor.middleware.large_int_id$convert_id_to_string$fn__47614.invoke(large_int_id.clj:59)","query_processor.middleware.format_rows$format_rows$fn__47595.invoke(format_rows.clj:74)","query_processor.middleware.add_default_temporal_unit$add_default_temporal_unit$fn__44219.invoke(add_default_temporal_unit.clj:23)","query_processor.middleware.desugar$desugar$fn__46917.invoke(desugar.clj:21)","query_processor.middleware.binning$update_binning_strategy$fn__45736.invoke(binning.clj:227)","query_processor.middleware.resolve_fields$resolve_fields$fn__46453.invoke(resolve_fields.clj:34)","query_processor.middleware.add_dimension_projections$add_remapping$fn__44574.invoke(add_dimension_projections.clj:312)","query_processor.middleware.add_implicit_clauses$add_implicit_clauses$fn__44803.invoke(add_implicit_clauses.clj:147)","query_processor.middleware.upgrade_field_literals$upgrade_field_literals$fn__49789.invoke(upgrade_field_literals.clj:40)","query_processor.middleware.add_source_metadata$add_source_metadata_for_source_queries$fn__45088.invoke(add_source_metadata.clj:123)","query_processor.middleware.reconcile_breakout_and_order_by_bucketing$reconcile_breakout_and_order_by_bucketing$fn__48942.invoke(reconcile_breakout_and_order_by_bucketing.clj:100)","query_processor.middleware.auto_bucket_datetimes$auto_bucket_datetimes$fn__45296.invoke(auto_bucket_datetimes.clj:147)","query_processor.middleware.resolve_source_table$resolve_source_tables$fn__46500.invoke(resolve_source_table.clj:45)","query_processor.middleware.parameters$substitute_parameters$fn__48750.invoke(parameters.clj:111)","query_processor.middleware.resolve_referenced$resolve_referenced_card_resources$fn__46552.invoke(resolve_referenced.clj:79)","query_processor.middleware.expand_macros$expand_macros$fn__47301.invoke(expand_macros.clj:184)","query_processor.middleware.add_timezone_info$add_timezone_info$fn__45097.invoke(add_timezone_info.clj:15)","query_processor.middleware.splice_params_in_response$splice_params_in_response$fn__49742.invoke(splice_params_in_response.clj:32)","query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__48953$fn__48957.invoke(resolve_database_and_driver.clj:31)","driver$do_with_driver.invokeStatic(driver.clj:60)","driver$do_with_driver.invoke(driver.clj:56)","query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__48953.invoke(resolve_database_and_driver.clj:25)","query_processor.middleware.fetch_source_query$resolve_card_id_source_tables$fn__47541.invoke(fetch_source_query.clj:274)","query_processor.middleware.store$initialize_store$fn__49751$fn__49752.invoke(store.clj:11)","query_processor.store$do_with_store.invokeStatic(store.clj:44)","query_processor.store$do_with_store.invoke(store.clj:38)","query_processor.middleware.store$initialize_store$fn__49751.invoke(store.clj:10)","query_processor.middleware.validate$validate_query$fn__49796.invoke(validate.clj:10)","query_processor.middleware.normalize_query$normalize$fn__47677.invoke(normalize_query.clj:22)","query_processor.middleware.add_rows_truncated$add_rows_truncated$fn__44943.invoke(add_rows_truncated.clj:35)","query_processor.middleware.results_metadata$record_and_return_metadata_BANG_$fn__49727.invoke(results_metadata.clj:143)","query_processor.middleware.constraints$add_default_userland_constraints$fn__46794.invoke(constraints.clj:42)","query_processor.middleware.process_userland_query$process_userland_query$fn__48839.invoke(process_userland_query.clj:135)","query_processor.middleware.catch_exceptions$catch_exceptions$fn__46734.invoke(catch_exceptions.clj:173)","query_processor.reducible$async_qp$qp_STAR___38051$thunk__38052.invoke(reducible.clj:103)","query_processor.reducible$async_qp$qp_STAR___38051$fn__38054.invoke(reducible.clj:108)"],"context":"json-download","error":"You do not have permissions to run this query.","row_count":0,"running_time":0,"ex-data":{"type":"missing-required-permissions","required-permissions":["/db/2/native/"],"actual-permissions":["/collection/7/read/","/db/2/schema/dbo/table/709/","/db/2/schema/dbo/table/949/","/collection/10/read/","/collection/4/","/collection/51/","/collection/51/read/","/collection/5/read/","/collection/14/read/","/collection/12/read/","/collection/2/read/","/db/2/schema/dbo/table/1350/","/collection/6/"],"permissions-error?":true},"data":{"rows":[],"cols":[]}}`
```
**To Reproduce**
1. Go to a saved native SQL query on a user without SQL query permissions.
2. Click and drag any column to reorganize it.
3. Click on "Download full results" and select any format.
4. See error in export.

**Expected behavior**
The export should happen without issue - ideally with the columns in the order that is displayed, but at worst the order of the original query.

**Information about your Metabase Installation:**

- Your browser and the version: Firefox 90.0b12
- Your operating system: Windows 10 21H1
- Your databases: SQL Server
- Metabase version: 0.39.4
- Metabase hosting environment: Docker (official image)
- Metabase internal database: PostgreDB

**Severity**
Low - can be worked around by either not reordering the columns or clicking back the "original" question before export.

### LINKED ISSUE #18433: Permissions issue where users are locked out of a question they have access to when editing filters
**Describe the bug**
Users that should have access to a question, can't access it after they change a filter a second time (the first time works).

**Logs**

```
[bbc10acb-ee33-4499-a187-105cea7a7b1a] 2021-10-13T15:43:16+10:30 ERROR metabase.query-processor.middleware.catch-exceptions Error processing query: null
{:database_id 2,
 :started_at #t "2021-10-13T15:43:16.277413+10:30[Australia/Adelaide]",
 :error_type :missing-required-permissions,
 :json_query
 {:database 2,
  :type "native",
  :native
  {:query
   "select \n    clients.id [[...redacted...]] limit 10",
   :collection "activities",
   :template-tags
   {:name
    {:id "28e10efd-bfaf-d063-593d-50c98c01788f", :name "name", :display-name "Name", :type "text", :required false},
    :email {:id "30c0930b-b463-2237-1159-5282047b6b31", :name "email", :display-name "Email", :type "text"},
    :user {:id "811e536c-b010-123c-aeab-b3c89c37305e", :name "user", :display-name "Mobile Number", :type "number"},
    :patient_id
    {:id "fe0a64ca-a010-c89c-706c-78645dbab927", :name "patient_id", :display-name "Patient", :type "number"}}},
  :parameters [{:type "category", :value "1232345", :target ["variable" ["template-tag" "patient_id"]]}],
  :middleware {:js-int-to-string? true, :add-default-userland-constraints? true}},
 :status :failed,
 :class clojure.lang.ExceptionInfo,
 :stacktrace
 ["--> query_processor.middleware.permissions$perms_exception.invokeStatic(permissions.clj:29)"
  "query_processor.middleware.permissions$perms_exception.doInvoke(permissions.clj:24)"
  "query_processor.middleware.permissions$perms_exception.invokeStatic(permissions.clj:26)"
  "query_processor.middleware.permissions$perms_exception.invoke(permissions.clj:24)"
  "query_processor.middleware.permissions$fn__45548$check_ad_hoc_query_perms__45553$fn__45554.invoke(permissions.clj:83)"
  "query_processor.middleware.permissions$fn__45548$check_ad_hoc_query_perms__45553.invoke(permissions.clj:79)"
  "query_processor.middleware.permissions$fn__45580$check_query_permissions_STAR___45585$fn__45586.invoke(permissions.clj:98)"
  "query_processor.middleware.permissions$fn__45580$check_query_permissions_STAR___45585.invoke(permissions.clj:88)"
  "query_processor.middleware.permissions$check_query_permissions$fn__45599.invoke(permissions.clj:107)"
  "query_processor.middleware.pre_alias_aggregations$pre_alias_aggregations$fn__50109.invoke(pre_alias_aggregations.clj:40)"
  "query_processor.middleware.cumulative_aggregations$handle_cumulative_aggregations$fn__45997.invoke(cumulative_aggregations.clj:60)"
  "query_processor.middleware.visualization_settings$update_viz_settings$fn__45935.invoke(visualization_settings.clj:63)"
  "query_processor.middleware.resolve_joined_fields$resolve_joined_fields$fn__47042.invoke(resolve_joined_fields.clj:102)"
  "query_processor.middleware.resolve_joins$resolve_joins$fn__50695.invoke(resolve_joins.clj:171)"
  "query_processor.middleware.add_implicit_joins$add_implicit_joins$fn__50231.invoke(add_implicit_joins.clj:190)"
  "query_processor.middleware.large_int_id$convert_id_to_string$fn__47061.invoke(large_int_id.clj:59)"
  "query_processor.middleware.format_rows$format_rows$fn__50283.invoke(format_rows.clj:74)"
  "query_processor.middleware.add_default_temporal_unit$add_default_temporal_unit$fn__46433.invoke(add_default_temporal_unit.clj:23)"
  "query_processor.middleware.desugar$desugar$fn__45908.invoke(desugar.clj:21)"
  "query_processor.middleware.binning$update_binning_strategy$fn__39130.invoke(binning.clj:229)"
  "query_processor.middleware.resolve_fields$resolve_fields$fn__45242.invoke(resolve_fields.clj:34)"
  "query_processor.middleware.add_dimension_projections$add_remapping$fn__49483.invoke(add_dimension_projections.clj:312)"
  "query_processor.middleware.add_implicit_clauses$add_implicit_clauses$fn__49781.invoke(add_implicit_clauses.clj:147)"
  "query_processor.middleware.upgrade_field_literals$upgrade_field_literals$fn__47429.invoke(upgrade_field_literals.clj:40)"
  "query_processor.middleware.add_source_metadata$add_source_metadata_for_source_queries$fn__46810.invoke(add_source_metadata.clj:123)"
  "query_processor.middleware.reconcile_breakout_and_order_by_bucketing$reconcile_breakout_and_order_by_bucketing$fn__50031.invoke(reconcile_breakout_and_order_by_bucketing.clj:100)"
  "query_processor.middleware.auto_bucket_datetimes$auto_bucket_datetimes$fn__48380.invoke(auto_bucket_datetimes.clj:147)"
  "query_processor.middleware.resolve_source_table$resolve_source_tables$fn__45223.invoke(resolve_source_table.clj:45)"
  "query_processor.middleware.parameters$substitute_parameters$fn__48034.invoke(parameters.clj:111)"
  "query_processor.middleware.resolve_referenced$resolve_referenced_card_resources$fn__45296.invoke(resolve_referenced.clj:79)"
  "query_processor.middleware.expand_macros$expand_macros$fn__51353.invoke(expand_macros.clj:184)"
  "query_processor.middleware.add_timezone_info$add_timezone_info$fn__47813.invoke(add_timezone_info.clj:15)"
  "query_processor.middleware.splice_params_in_response$splice_params_in_response$fn__50298.invoke(splice_params_in_response.clj:32)"
  "query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__49792$fn__49796.invoke(resolve_database_and_driver.clj:31)"
  "driver$do_with_driver.invokeStatic(driver.clj:60)"
  "driver$do_with_driver.invoke(driver.clj:56)"
  "query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__49792.invoke(resolve_database_and_driver.clj:25)"
  "query_processor.middleware.fetch_source_query$resolve_card_id_source_tables$fn__45846.invoke(fetch_source_query.clj:274)"
  "query_processor.middleware.store$initialize_store$fn__46034$fn__46035.invoke(store.clj:11)"
  "query_processor.store$do_with_store.invokeStatic(store.clj:44)"
  "query_processor.store$do_with_store.invoke(store.clj:38)"
  "query_processor.middleware.store$initialize_store$fn__46034.invoke(store.clj:10)"
  "query_processor.middleware.validate$validate_query$fn__50081.invoke(validate.clj:10)"
  "query_processor.middleware.normalize_query$normalize$fn__50088.invoke(normalize_query.clj:22)"
  "query_processor.middleware.add_rows_truncated$add_rows_truncated$fn__47759.invoke(add_rows_truncated.clj:35)"
  "query_processor.middleware.results_metadata$record_and_return_metadata_BANG_$fn__49057.invoke(results_metadata.clj:147)"
  "query_processor.middleware.constraints$add_default_userland_constraints$fn__47777.invoke(constraints.clj:42)"
  "query_processor.middleware.process_userland_query$process_userland_query$fn__50067.invoke(process_userland_query.clj:146)"
  "query_processor.middleware.catch_exceptions$catch_exceptions$fn__50378.invoke(catch_exceptions.clj:169)"
  "query_processor.reducible$async_qp$qp_STAR___42481$thunk__42482.invoke(reducible.clj:103)"
  "query_processor.reducible$async_qp$qp_STAR___42481.invoke(reducible.clj:109)"
  "query_processor.reducible$sync_qp$qp_STAR___42490$fn__42493.invoke(reducible.clj:135)"
  "query_processor.reducible$sync_qp$qp_STAR___42490.invoke(reducible.clj:134)"
  "query_processor$process_userland_query.invokeStatic(query_processor.clj:241)"
  "query_processor$process_userland_query.doInvoke(query_processor.clj:237)"
  "query_processor$fn__51401$process_query_and_save_execution_BANG___51410$fn__51413.invoke(query_processor.clj:253)"
  "query_processor$fn__51401$process_query_and_save_execution_BANG___51410.invoke(query_processor.clj:245)"
  "query_processor$fn__51445$process_query_and_save_with_max_results_constraints_BANG___51454$fn__51457.invoke(query_processor.clj:265)"
  "query_processor$fn__51445$process_query_and_save_with_max_results_constraints_BANG___51454.invoke(query_processor.clj:258)"
  "api.dataset$run_query_async$fn__57767.invoke(dataset.clj:63)"
  "query_processor.streaming$streaming_response_STAR_$fn__38039$fn__38040.invoke(streaming.clj:160)"
  "query_processor.streaming$streaming_response_STAR_$fn__38039.invoke(streaming.clj:159)"
  "async.streaming_response$do_f_STAR_.invokeStatic(streaming_response.clj:65)"
  "async.streaming_response$do_f_STAR_.invoke(streaming_response.clj:63)"
  "async.streaming_response$do_f_async$fn__25228.invoke(streaming_response.clj:84)"],
 :card_id nil,
 :context :ad-hoc,
 :error "You do not have permissions to run this query.",
 :row_count 0,
 :running_time 0,
 :ex-data
 {:type :missing-required-permissions,
  :required-permissions #{"/db/2/native/"},
  :actual-permissions
  #{"/collection/291/"
    "/db/10/schema/"
    "/collection/258/"
    "/collection/84/"
    "/collection/282/"
    "/collection/249/read/"
    "/collection/279/"
    "/collection/238/"
    "/collection/74/read/"
    "/collection/39/"
    "/collection/244/"
    "/collection/284/"
    "/collection/259/"
    "/collection/1/"
    "/collection/240/"
    "/collection/237/"
    "/collection/285/"
    "/collection/268/"
    "/collection/267/"
    "/collection/255/"
    "/collection/251/"
    "/collection/272/"
    "/collection/3/"
    "/collection/290/"
    "/collection/253/"
    "/collection/262/"
    "/collection/292/"
    "/collection/248/read/"
    "/collection/293/"
    "/db/2/schema/"
    "/collection/241/"
    "/collection/269/"
    "/collection/295/"
    "/collection/261/read/"
    "/collection/273/"
    "/collection/287/"
    "/db/13/"
    "/collection/294/"
    "/collection/270/"
    "/collection/7/"
    "/collection/276/"
    "/collection/109/"
    "/collection/76/"
    "/collection/141/"
    "/collection/266/"
    "/collection/275/"
    "/collection/264/"
    "/collection/265/"
    "/collection/278/"
    "/collection/242/"
    "/collection/42/"
    "/collection/263/"
    "/collection/256/read/"
    "/collection/239/"
    "/collection/252/"
    "/collection/40/"
    "/collection/289/"
    "/collection/288/"
    "/collection/280/"
    "/collection/274/"},
  :permissions-error? true},
 :data {:rows [], :cols []}}
```

From the logs, I can see that the query was ran as a native query, even though it was accessed from a question, without any modifications other than changing a filter.

**To Reproduce**
1. Create a question with a filter in a collection
2. Create a user and give them access to only that collection
3. Go to the question as that user, fill in the filter and press search
4. Edit the filter and press search again
5. Get an error "You do not have permissions to run this query."
6. The URL at the top after the second search has a hash (which seems to indicate that it ran a modified raw query). The URL after the first search doesn't have a hash, just a `?param` added at the end.

**Expected behavior**
Both searches work correctly as the user has access to the question.
This is a regression, as it used to work fine in 0.38 (maybe even after that).

**Information about your Metabase Installation:**
- Your browser and the version: Chrome 94
- Your operating system: Ubuntu 21.04
- Your databases: MySQL
- Metabase version: 0.41.0
- Metabase hosting environment: Docker
- Metabase internal database: MySQL

```
{
  "browser-info": {
    "language": "en",
    "platform": "Linux x86_64",
    "userAgent": "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/94.0.4606.81 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.12+7",
    "java.vendor": "Eclipse Foundation",
    "java.vendor.url": "https://adoptium.net/",
    "java.version": "11.0.12",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.12+7",
    "os.name": "Linux",
    "os.version": "4.14.239-aptible.0",
    "user.language": "en",
    "user.timezone": "Australia/Adelaide"
  },
  "metabase-info": {
    "databases": [
      "mysql",
      "googleanalytics"
    ],
    "hosting-env": "unknown",
    "application-database": "mysql",
    "application-database-details": {
      "database": {
        "name": "MySQL",
        "version": "8.0.25"
      },
      "jdbc-driver": {
        "name": "MariaDB Connector/J",
        "version": "2.6.2"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2021-10-06",
      "tag": "v0.41.0",
      "branch": "release-x.41.x",
      "hash": "c529fe2"
    },
    "settings": {
      "report-timezone": "Australia/Adelaide"
    }
  }
}
```

**Severity**
There is a work around - you can go back to the question and do a new search and that works, but it does get annoying and might force us to downgrade to mitigate the issue.

**Additional context**
It's a regression - it used to work fine in 0.38

### PR DESCRIPTION
**Pull request by kulyk**
Fixes #16836, fixes #18433

Continuation of #20317 and #20320. This PR would allow no-data users to:

* resize columns and change their order for table visualization
* change a question's visualization type and visualization settings

Both things would only work "locally", i.e. there will be no "Save" button to keep the changes.

### To Verifty

1. Create a question visualized as a table
2. Sign in as a no-data user
3. Try to resize/rearrange columns
4. Ensure it worked, ensure the URL is not changed to `/question#longHashValue`, ensure you don't see the "Save" button
5. Try to change visualization type and viz settings
6. Ensure it worked, ensure the URL is not changed to `/question#longHashValue`, ensure you don't see the "Save" button
7. Refresh the page, ensure the question looks the same as originally (without the changes you've made)

### Demo

https://user-images.githubusercontent.com/17258145/164306279-a0180327-96da-4e22-8547-f5ae88326555.mp4

### CODE REVIEW

**Review by flamber — APPROVED** [review]
@kulyk Excellent.

Yes it fixes #16836.

Though, I find it slightly strange that I can change column names and other vizsettings, and not given any options to save, and those changes are not reflected in the exports.
Perhaps we should only allow humans changing the column width, while allow other things underneath to fix the linked issues?

----

It also almost fixes #18433, though only on first filter definition and then it causes gray-out (possibly the Run-overlay, but just with hidden Run-button?)

https://user-images.githubusercontent.com/1447303/164896500-c687e6d9-dc41-4536-bd0d-1f35c2ab8f2e.mp4

**Review by daltojohnso — APPROVED** [review]
👏

**Comment by kulyk** [comment]
@flamber thanks for your review! Could you please help me to clarify a couple of things:

> I find it slightly strange that I can change column names and other vizsettings, and not given any options to save, and those changes are not reflected in the exports.
>
> Perhaps we should only allow humans changing the column width, while allow other things underneath to fix the linked issues?

But does it make sense to show the "Save" button to no-data users at all? Any attempt to save the changes will lead to "you don't have permission" error. Or did I accidentally hide the "Save" button in cases it's actually needed?

> It also almost fixes #18433, though only on first filter definition and then it causes gray-out (possibly the Run-overlay, but just with hidden Run-button?)

I'll look into that!

File: frontend/test/metabase/scenarios/question/new.cy.spec.js [hunk 1]
```diff
@@ -367,31 +367,6 @@ describe("scenarios > question > new", () => {
...
-
-          cy.findByText("Started from").click(); // Give DOM some time to update
-
-          cy.get("@headerCell").then($newCell => {
-            const newWidth = $newCell[0].getBoundingClientRect().width;
-
-            expect(newWidth).to.be.gt(originalWidth);
-          });
-        });
-    });
```
[the remark this answers is not in the document]
Comment (nemanjaglumac) [reply frontend/test/metabase/scenarios/question/new.cy.spec.js:394]: Something like this, if it's not too hacky?
```js
const originalWidth = $cell[0].getBoundingClientRect().width;

cy.wrap($cell)
  .find(".react-draggable")
  .trigger("mousedown", 0, 0, { force: true })
  .trigger("mousemove", 100, 0, { force: true })
  .trigger("mouseup", 100, 0, { force: true });

foo();

function foo(index = 0) {
  cy.get("@headerCell").then($newCell => {
    const newWidth = $newCell[0].getBoundingClientRect().width;

    if (newWidth === originalWidth && index < 3) {
      cy.wait(100);
      foo(++index);
    } else {
      expect(newWidth).to.be.gt(originalWidth);
    }
  });
}
```

Worked for me locally.

[the remark this answers is not in the document]
Comment (kulyk) [reply frontend/test/metabase/scenarios/question/new.cy.spec.js:394]: Thanks @nemanjaglumac, I'll try it out!

**Comment by flamber** [comment]
@kulyk I think we've had so much inconsistency over the years, which has just made it difficult for me to figure out what the plan is anymore. I don't see the point in showing how to change the visualization or it's settings, when you cannot save it anyway.
Wouldn't you expect to be able to save if you can make a lot of change to something?
It's not important - your PR is miles better than how it currently functions. We do not want all the permission errors anymore.

**Review by flamber — APPROVED** [review]
I'm not sure what is needed to solve this, but if I make a viz change with nodata and run the query with Ctrl+Enter, then it shows permission error, and using browser back-button shows completely blank page.

https://user-images.githubusercontent.com/1447303/165312864-371542d6-a3d4-441a-8f20-aa95fe92ed2f.mp4

Let's merge this - I think the correct solution is to not allow any settings at all - if we want to allow resize of columns for nodata, then we can open a feature request for that.

**Pull request merged**